### PR TITLE
Attacking the fire alarm with an item now turns it on

### PR DESCRIPTION
**Pull request by nicbn**
[Changelogs]: #
:cl: nicbn
tweak: Attacking the fire alarm with anything that has force and won't deconstruct it will turn it on
fix: Toggling the fire alarm now leaves fingerprints
/:cl:

### CODE REVIEW

**Comment by FantasticFwoosh** [comment]
I understand the concept but in practice this sounds very annoying, could you please make it rely on a nessecary force factor so bees & simple mobs hitting walls idly dont activate it per-instance?

**Comment by swankcookie** [comment]
> skateboard into fire alarm accidentally while clown is nearby
>fire doors slide shut
>suddenly cage match

**Comment by nicbn** [comment]
This only happens if you attack it with an object that has force.

**Comment by epochayur** [comment]
what about shooting it? the idea of being able to pop a shot at an alarm and slip through the fire doors before they close to evade pursuers would be the tits

File: code/game/machinery/firealarm.dm [hunk 1]
```diff
@@ -131,23 +131,23 @@
...
 		alarm()
 
+/obj/machinery/firealarm/attack_hand(mob/user)
+	if(buildstage != 2)
+		return ..()
+	add_fingerprint(user)
+	toggle()
+
 /obj/machinery/firealarm/attack_ai()
 	if(buildstage != 2)
```
[the remark this answers is not in the document]
Comment (nicbn) [reply code/game/machinery/firealarm.dm]: There is no `user` in `attack_ai`. Also `attack_hand` adds a fingerprint.

[the remark this answers is not in the document]
Comment (ninjanomnom) [reply code/game/machinery/firealarm.dm]: Yes there is, this version just doesn't have the arg shown but it's still passed in

[the remark this answers is not in the document]
Comment (nicbn) [reply code/game/machinery/firealarm.dm]: Ok, so calling `attack_hand` will add an unwanted fingerprint

**Pull request merged**